File: TP1 - C++/data/models/expense.cpp
#include "../../domain/entities/expense.hpp"

Expense::Expense(string id, string desciption, string dueDate, string paymentDate, string expenseType, float value)
{   
    this->value = value;
    this->id = id;
    this->description = description;
    this->dueDate = dueDate;
    this->paymentDate = paymentDate;
    this->expenseType = expenseType;
};

Expense::~Expense(){};

string Expense::getId()
{
    return this->id;
}

string Expense::getDescription()
{
    return this->description;
}

string Expense::getDueDate()
{
    return this->dueDate;
}

string Expense::getPaymentDate()
{
    return this->paymentDate;
}

string Expense::getExpenseType()
{
    return this->expenseType;
}

float Expense::getValue(){
    return this->value;
}

void Expense::setDescription(string description)
{
    this->description = description;
}

void Expense::setDueDate(string dueDate)
{
    this->dueDate = dueDate;
}

void Expense::setPaymentDate(string paymentDate)
{
    this->paymentDate = paymentDate;
}

void Expense::setExpenseType(string expenseType)
{
    this->expenseType = expenseType;
}

void Expense::setValue(float value){
    this->value = value;
}

File: TP2 - Java/src/domain/entities/Expense.java
package domain.entities;

public class Expense {
  private final String id;
  private String description;
  private String dueDate;
  private String paymentDate;
  private float value;

  public Expense(String id, String description, String dueDate, String paymentDate, float value) {
    this.id = id;
    this.description = description;
    this.dueDate = dueDate;
    this.paymentDate = paymentDate;
    this.value = value;
  }

  public String getId() {
    return id;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getDueDate() {
    return dueDate;
  }

  public void setDueDate(String dueDate) {
    this.dueDate = dueDate;
  }

  public String getPaymentDate() {
    return paymentDate;
  }

  public void setPaymentDate(String paymentDate) {
    this.paymentDate = paymentDate;
  }

  public float getValue() {
    return value;
  }

  public void setValue(float value) {
    this.value = value;
  }
}
